src/dataman/RowReader.cpp

```diff
@@ -244,14 +244,14 @@ bool RowReader::processHeader(folly::StringPiece row) {
     }
     offsets_.resize(numFields + 1, -1);
     offsets_[0] = 0;
-    blockOffsets_.push_back(std::make_pair(0, 0));
+    blockOffsets_.emplace_back(std::make_pair(0, 0));
```

Comment:
You should have known that `std::make_pair` is not necessary here if `emplace_back` used.

Reply:
Very careful review . I benefit greatly by this point. You are the master of c++. 
Let me address them just now.